deployments/boringssl-envoy-deployment.yaml

```diff
@@ -108,7 +115,23 @@ spec:
             - name: config
               mountPath: /etc/envoy/config
               readOnly: true
+            - name: workdir
+              mountPath: "/usr/local/bin"
+      initContainers:
+        - name: init-config
+          image: busybox
+          volumeMounts:
+          - name: workdir
+            mountPath: "/usr/share"
+          command: ['/bin/sh', '-c']
+          args:
+            - "echo '#!/bin/sh\n
+            set -e\n
+            /usr/bin/envsubst < /etc/envoy/config/envoy-conf.yaml > /tmp/envoy-conf.yaml\n
+            /envoy-static -c /tmp/envoy-conf.yaml $@' > /usr/share/entrypoint.sh"
```

Comment:
You shouldn't run Envoy from the initContainer; just create the configuration file in it and let the initContainer run to completition. And I'm not even sure if initContainer is a better solution than doing an entrypoint script for the regular container -- what do you think?

Reply:
@ipuustin This init container only do that, write the config file in a volume, and then is terminated, after init container died envoy container starts but now the entrypoint.sh file is loaded in the right path to be executed by the entrypoint specified in the dockerfile.  But as same as you, I'm not quite sure of this approach is better, init adds few time more in starting  (not so much since is busybox and that image is only 1.2MB)...  at least with the dockerfile once the images is built we are sure that the entrypoint will be there and will be working... I haven't see another interesting approach, so I think I will move back to the Dockerfile is OK??

Reply:
Ok, I think I just read the diff wrong. Sorry about the noise. :-) If you think going back to the one container approach makes more sense, please go ahead.